pkg/sbproxy/sbproxy.go

```diff
@@ -0,0 +1,227 @@
+package sbproxy
+
+import (
+	"context"
+	"strconv"
+	"sync"
+
+	"os"
+	"os/signal"
+
+	"time"
+
+	"fmt"
+
+	"net/http"
+
+	"github.com/Peripli/service-broker-proxy/pkg/osb"
+	"github.com/Peripli/service-broker-proxy/pkg/platform"
+	"github.com/Peripli/service-broker-proxy/pkg/sbproxy/middleware"
+	"github.com/Peripli/service-broker-proxy/pkg/sm"
+	"github.com/Peripli/service-broker-proxy/pkg/task"
+	"github.com/gorilla/mux"
+	"github.com/pmorie/osb-broker-lib/pkg/metrics"
+	"github.com/pmorie/osb-broker-lib/pkg/rest"
+	"github.com/pmorie/osb-broker-lib/pkg/server"
+	prom "github.com/prometheus/client_golang/prometheus"
+	"github.com/robfig/cron"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	BrokerPathParam = "brokerID"
+	ApiPrefix       = "/v1/osb/{" + BrokerPathParam + "}"
+)
+
+var (
+	ctx    context.Context
+	cancel context.CancelFunc
+	group  sync.WaitGroup
+)
+
+type Configuration interface {
+	Validate() error
+	SbproxyConfig() *ServerConfiguration
+	OsbConfig() *osb.ClientConfiguration
+	SmConfig() *sm.ClientConfiguration
+	PlatformConfig() platform.ClientConfiguration
+}
```

Comment:
why not use a simple struct instead of this interface?
what is the reason for `Validate` method? Configuration could be validated at the point of creation

Reply:
because we need to abstract away the creation of the platform client - notice that it is interface and is plugged by the wrapper

Reply:
On seconds thoughts i think we can go for a  simple  struct for the big config - let me try and see how it looks